File: Java/PdfViewerAPI_R2/java/com/pdf_tools/pdfviewer/Requests/PdfGetPageLayoutRequest.java
package com.pdf_tools.pdfviewer.Requests;

import com.pdf_tools.pdfviewer.DocumentManagement.IPdfDocument;
import com.pdf_tools.pdfviewer.Model.IPdfControllerCallbackManager;
import com.pdf_tools.pdfviewer.Model.PdfViewerException;
import com.pdf_tools.pdfviewer.Model.IPdfCommon.TPageLayoutMode;

public class PdfGetPageLayoutRequest extends APdfRequest<TPageLayoutMode>
{

    public PdfGetPageLayoutRequest()
    {
        this._priority = 85;
    }

    @Override
    public void execute(IPdfDocument document, IPdfControllerCallbackManager controller)
    {

        try
        {
            TPageLayoutMode layout = document.getPageLayout();
            completedEvent.triggerEvent(layout, null);
        } catch (PdfViewerException ex)
        {
            completedEvent.triggerEvent(null, ex);
        }
    }

}

File: CS.NET/PdfViewerCSharpAPI_R2/DocumentManagement/Requests/PdfGetPageLayoutRequest.cs
// -----------------------------------------------------------------------
// <copyright file="PdfGetLayoutModeRequest.cs" company="">
// TODO: Update copyright text.
// </copyright>
// -----------------------------------------------------------------------

namespace PdfTools.PdfViewerCSharpAPI.DocumentManagement.Requests
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;
    using PdfTools.PdfViewerCSharpAPI.Utilities;
    using PdfTools.PdfViewerCSharpAPI.Model;

    /// <summary>
    /// TODO: Update summary.
    /// </summary>
    public class PdfGetPageLayoutRequest : APdfRequest<object, TPageLayoutMode>
    {
        public PdfGetPageLayoutRequest()
            :base(null, 85)
        {
        }

        protected override TPageLayoutMode ExecuteNative(IPdfDocument document, object nothing)
        {
            return document.GetPageLayout();
        }

        protected override void triggerControllerCallback(IPdfControllerCallbackManager controller, InOutTuple tuple, PdfViewerException ex)
        {
        }
        protected override void triggerControllerCallback(IPdfControllerCallbackManager controller, PdfViewerException ex)
        {
        }
    }
}
